blockdag/validate.go

```diff
@@ -545,6 +545,22 @@ func (dag *BlockDAG) checkBlockSanity(block *util.Block, flags BehaviorFlags) (t
 		existingTxIDs[*id] = struct{}{}
 	}
 
+	// Check for duplicate transactions. This check will be fairly quick
+	// since the transaction IDs are already cached due to building the
+	// merkle tree above.
+	usedOutpoints := make(map[wire.Outpoint]struct{})
```

Comment:
Consider the map being from outpoint to tx that spends it.
This way you can print in the error message which is the doublespending transaction - might save a tons of debugging to somebody in the future